Re write obcs tides

jm-c opened the pull request:
## What changes does this PR introduce?
Implement suggestions from issue #617, renaming OB tidal forcing variables and improving
efficiency of tidal component addition. Also allows to add tides to OB tangential velocity.

## What is the current behaviour?

1. not very efficient (see point 1, issue #617 )
2. not very flexible regarding number of tidal-components to set in binary input files (see point 2, issue #617 )
3. some confusing names, with "normal/tangential" velocity component used for Tides but "u/v" component used for the remaining OBCS pkg (see point 3, issue #617 )

## What is the new behaviour 
Address all 3 points above.
Also enable to specify the tangential tidal velocity at the OB (An other implementation of PR #73).

## Does this PR introduce a breaking change? 
Names of Tidal forcing parameters have changed. A check and stop has been added to `obcs_readparams.F` if old-names were found in `data.obcs`.

## Other information:
1. loading and setting of OB tidal forcing has been moved from `obcs_init_variables.F` to `obcs_init_fixed.F` since the tidal components amplitude and phase are kept constant during one simulation.
2. point (2) above is now implemented (using updated code from PR #747 which has now been merged in).
3. the new (more efficient) calculation of tidal flow introduces differences at machine-truncation level ;
  a temporary CPP option "OLD_OBCS_TIDES" was allowing to recover the original results (but got removed).
4. The updated example test `seaice_obcs.tides` from PR #73 has been moved here with minor renaming and smaller binary files (since only 4 components are used). 

## Suggested addition to `tag-index`
o pkg/obcs (tides):
  - more efficient implementation of OB barotropic tidal velocity components,
    with both Cos & Sin of phase times Amplitude stored in common block ;
  - more flexible specification of OB tidal-components (only read-in components
    with non-zero tidal period) ;
  - allows to specify tangential component of barotropic tidal velocity at OB ;
  - rename all OBC tidal component input parameter (period, amplitude and
    phase of each OB barotropic tidal velocity component input files) ;
  - upgrade OB tidal example (in seaice_obcs/input.tides) with fewer tidal
    components (only 4) but including few tangential components.

jm-c commented:
Two comments:

1. I would like to remove completely the CPP option `OLD_OBCS_TIDES` that I added in this branch once we are happy with the new implementation. So we don't have to document this new CPP option.
2. the reference output for the `seaice_obcs` test with tides (`output.tides.txt`) has been imported from PR #73, and it shows that this test passes. Might update this ref. output before merging this PR into master branch, so that it will be cleaner.

antnguyen13 commented:
A quick q: generally we either use amp & phase or split into sin and cos components , e.g., y = Acos(2pi/T t) + Bsin(2pi/T t) = sqrt(A^2+B^2) cos (2pi/T t - tan^{-1}(B/A)) .  Am I correct you're switching from the latter (old, am, ph) to the former (cos and sin)?

Also, in the "old" way of am/ph, it was only for the "normal" component, e.g., if eastern boundary, we get u = normal.  What you're redoing now, not only to use the sin and cos form (thus getting away from am, ph), but that there'll be an additional sin + cos for the tangential component? Thus the way you've coded now, with cos+sin for u and same for v, naturally one will be "normal" and one will be tangential, right? Thus bypassing the mod code of PR where the "t" am and ph were added?

jm-c commented:
@antnguyen13 Thanks for reviewing this and regarding you questions:
> A quick q: generally we either use amp & phase or split into sin and cos components , e.g., y = Acos(2pi/T t) + Bsin(2pi/T t) = sqrt(A^2+B^2) cos (2pi/T t - tan^{-1}(B/A)) . Am I correct you're switching from the latter (old, am, ph) to the former (cos and sin)?
> 
> Also, in the "old" way of am/ph, it was only for the "normal" component, e.g., if eastern boundary, we get u = normal. What you're redoing now, not only to use the sin and cos form (thus getting away from am, ph), but that there'll be an additional sin + cos for the tangential component? Thus the way you've coded now, with cos+sin for u and same for v, naturally one will be "normal" and one will be tangential, right? Thus bypassing the mod code of PR#73 where the "t" am and ph were added?

You are right, and this match this PR description:
> Implement suggestions from issue https://github.com/MITgcm/MITgcm/issues/617, renaming OB tidal forcing variables and improving
> efficiency of tidal component addition. Also allows to add tides to OB tangential velocity.

And to reiterate, we continue to specify (in input files) the amplitude and phase (no changes here) but we store (in common block) the pair product: Amplitude x cos,sin [2*pi*Phase/Period]

jm-c commented on pkg/obcs/obcs_add_tides.F:
```diff
@@ -47,83 +43,174 @@ SUBROUTINE OBCS_ADD_TIDES( myTime, myIter, myThid )
-     &              maskW(iB,j,k,bi,bj) * OBEam(j,td,bi,bj) *
-     &              COS( 2.D0 * PI * (myTime-OBEph(j,td,bi,bj)) /
-     &                   tidalPeriod(td) )
+     &              maskW(iB,j,k,bi,bj) * OBE_uTideCs(j,td,bi,bj) *
```
The lines you are pointed to are within "#ifdef  OLD_OBCS_TIDES"  (lines 63 - 124) and I don't disagree with you. This is the reason I added this comment in Aug 22, 2023:
> Two comments:
>   I would like to remove completely the CPP option OLD_OBCS_TIDES that I added in this branch once we are happy with the new implementation. So we don't have to document this new CPP option.

mjlosch replied:
If you only read this part of the code, then these line appear to make no sense. But the fields are in "dual-use" and if `OLD_OBCS_TIDES` is defined, they have a different meaning: then  `OB?_uTideCs = OB?am` and `OBN_uTideSn = OB?ph` and this corresponds to what one would expect.

I think the only way to avoid the admittedly confusing code is to keep variables with the old names just for the `OLD_OBCS_TIDES` code. Doing so only makes sense (too me), if we plan on keeping the `OLD_OBCS_TIDES` code for bitwise reproducibility. But as I understand, this is not the plan. 

The way I understand this PR, one can reproduce the old results by making sure that `OB?_uTideCs = OB?_vTideCs =OB?am` and `OBN_uTideSn = OBN_vTideSn = OB?ph` by specifying the appropriate input fields (subject to roundoff error), so the `OLD_OBCS_TIDES` code is not necessary and can be removed. This would also remove the confusion.

Neither the old code nor the new code is yet properly documented anywhere. I'd vote for removing `OLD_OBCS_TIDES` code and document the new code properly (and maybe how to recover old results with the new code).

mjlosch commented on pkg/obcs/OBCS_PARAMS.h:
```diff
@@ -166,9 +168,11 @@ C insideOBmaskFile   :: File to specify Inside OB region mask (zero beyond OB)
-     &      OBNamFile, OBSamFile, OBEamFile, OBWamFile,
-     &      OBNphFile, OBSphFile, OBEphFile, OBWphFile,
+     &      OBNetaFile, OBSetaFile, OBEetaFile, OBWetaFile,
+     &  OBN_uTidAmFile, OBS_uTidAmFile, OBE_uTidAmFile, OBW_uTidAmFile,
+     &  OBN_vTidAmFile, OBS_vTidAmFile, OBE_vTidAmFile, OBW_vTidAmFile,
+     &  OBN_uTidPhFile, OBS_uTidPhFile, OBE_uTidPhFile, OBW_uTidPhFile,
+     &  OBN_vTidPhFile, OBS_vTidPhFile, OBE_vTidPhFile, OBW_vTidPhFile,
```
Is there a good reason for the truncated naming? Why not call them `OBN_uTideAmpltFile` and `OBN_uTidePhaseFile` or `OBN_uTidalAmpltFile` and `OBN_uTidalPhaseFile`etc.

jm-c replied:
All the other reference to field name within OB file-name parameter are short or very short (e.g., "u" instead of "uVel", "S" instead of "Salt"), so I would prefer to keep these ones short too (as were the previous names, with "am" and "ph" for Amplitude and Phase).

mjlosch reviewed (approved):
In general, the changes look good to me. I have these comments:
- At first sight, it is complicated to relate the new code the old one, but that's maybe not a big problem. 
- Related to the previous comment: With this PR the results of `seaice_obcs.tides` are very different. This may be intentional, but would it be helpful to create results that would reproduce results of the old code? I tried by using the old `OB?am/ph.seaice_obcs` and adjusting `data.obc` like this: 
```
 OBCS_tidalPeriod  = 44714.16,43200.,45569.88,43081.92,86164.2,92949.48,86637.24,96726.24,1180295.64,2380716.,
 OBS_vTidamFile='OBSam.seaice_obcs',
 OBS_vTidphFile='OBSph.seaice_obcs',
 OBN_vTidamFile='OBNam.seaice_obcs',
 OBN_vTidphFile='OBNph.seaice_obcs',
 OBE_uTidamFile='OBEam.seaice_obcs',
 OBE_uTidphFile='OBEph.seaice_obcs',
 OBW_uTidamFile='OBWam.seaice_obcs',
 OBW_uTidphFile='OBWph.seaice_obcs',
 useOBCStides = .TRUE.,
``` 
and got nearly identical results (13 matching digits on my MacBook, when it was 14 with upstream/master).
- If we can illustrate or describe somewhere how one can reproduce the old results with the new code without the cpp-flag `OLD_OBCS_TIDES`, then we can certainly get rid off this cpp-flag and the corresponding code (and should).

mjlosch commented:
I think it is enough to document somewhere (maybe in `seaice_obcs/README`, which could also be renamed to `README.md` to make it a markdown file, or even in `seaice_obcs/input.tides/data.obcs` itself) that with this PR everything (namelist names, variable names) changed, but that it is possible to recover the old results obtained with old code and
```
 useOBCStides = .TRUE.,
#components   = M2       S2     N2       K2       K1      O1       P1       Q1       Mf         Mm
#periods (hr) = 12.4206  12     12.6583  11.9672  23.9345 25.8193  24.0659  26.8684  327.8599   661.31
 tidalPeriod  = 44714.16,43200.,45569.88,43081.92,86164.2,92949.48,86637.24,96726.24,1180295.64,2380716.,
 OBSamFile='OBSam.seaice_obcs',
 OBNamFile='OBNam.seaice_obcs',
 OBEamFile='OBEam.seaice_obcs',
 OBWamFile='OBWam.seaice_obcs',
 OBSphFile='OBSph.seaice_obcs',
 OBNphFile='OBNph.seaice_obcs',
 OBEphFile='OBEph.seaice_obcs',
 OBWphFile='OBWph.seaice_obcs',
```
by new code and
```
 useOBCStides = .TRUE.,
#components   = M2       S2     N2       K2       K1      O1       P1       Q1       Mf         Mm
#periods (hr) = 12.4206  12     12.6583  11.9672  23.9345 25.8193  24.0659  26.8684  327.8599   661.31
 OBCS_tidalPeriod  = 44714.16,43200.,45569.88,43081.92,86164.2,92949.48,86637.24,96726.24,1180295.64,2380716.,
 OBS_vTidamFile='OBSam.seaice_obcs',
 OBS_vTidphFile='OBSph.seaice_obcs',
 OBN_vTidamFile='OBNam.seaice_obcs',
 OBN_vTidphFile='OBNph.seaice_obcs',
 OBE_uTidamFile='OBEam.seaice_obcs',
 OBE_uTidphFile='OBEph.seaice_obcs',
 OBW_uTidamFile='OBWam.seaice_obcs',
 OBW_uTidphFile='OBWph.seaice_obcs',
```
This would provide some sort of recipe for recovering old results in different contexts, if required.
And we do not need to keep the `*.seaice_obcs` files. If necessary, they can always be recovered from an old checkpoint.

jm-c commented:
@mjlosch I like your suggestion, I will do one or the other (REAME or data.obcs). Also will remove completely `OLD_OBCS_TIDES` code.

jm-c commented:
@mjlosch I removed all the `OLD_OBCS_TIDES` code and add comment in README with reference to a small file for updating `data.obcs` content. You can check if it's ok.
@antnguyen13 I think I addressed your comments, and if not, please let us know.

And @menemenlis we are still waiting for you approval (you seems to agreee with issue #617, and this PR is mostly implementing that).

mjlosch commented:
I'll push small modifications to the `README.me`. I can also add some links files in the read (e.g. `input.tides/update_TideFileName.sed`), so one can click on this on the web-page, if desired.

jm-c commented:
@mjlosch the README.md now looks even better ! thanks.

menemenlis commented:
It looks good to me, ready for merge, but a few minor comments for your consideration:

1. perhaps add a little more detail in MITgcm/verification/seaice_obcs/README.md, for example:

Note: naming of tidal input files (OB*File) have been changed and augmented to accomodate tangential flows in PR [#752](https://github.com/MITgcm/MITgcm/pull/752)
 (see [`input.tides/update_TideFileName.sed`](input.tides/update_TideFileName.sed) on how to update `data.obcs` to map the old normal flow components to the new names)

2. perhaps add a comment in OBCS_FIELDS.h and obcs_init_fixed.F to make clear that OB[N,S,E,W]_[u,v]Tide[Cs,Sn] arrays contain amplitude and phase during initialization before being converted to Sine and Cosine, for example:

C Note that during initialization, the Cs arrays contain Amplitude
C                              and the Sn arrays contain Phase

jm-c commented:
@menemenlis Thanks for the review. I made little adjustment to the README.md file in `seaice_obcs` following your suggestion. 
Regarding your second point, since the only place where 
> the Cs arrays contain Amplitude and the Sn arrays contain Phase

is locally in `obcs_init_fixed.F`, it might add more confusion if this comment is found in `OBCS_FIELDS.h`. And in `obcs_init_fixed.F` we already have:
> 454       IF ( useOBCStides ) THEN
> 455 C--   Read from files Barotropic Tidal Amplitude and Phase:
...
> 536 C--   Set Tidal coeff (= Amplit x COS & SIN of Phase ) from Amplitude and Phase:

so it's not obvious to me that we need to repeat/re-phrase it there. So at the end, would prefer to leave it as it is now.

jm-c commented:
I've jsut updated this PR description (including suggested addition to tag-index) so, from my side, this PR is ready to be merged (as I understood, @menemenlis approval extends to @antnguyen13 approval) and will merge it in the coming days  unless someone wants to add something.

kartadikaria commented:
Dear All,

Can I set useEXF=.FALSE., but I still use useOBCStides =.TRUE.
and useOBCSprescribe=.TRUE. in my data.obcs? I need to reduce CFL by
reducing deltaT < 1. Since data.exf can't use deltaT < 1, then I turn off
data.exf. However, I still want to read my prescribed OBCS and tides OBCS
data in data.obcs. Is that possible?

How to set periodicExternalForcing, externForcingPeriod and
externForcingCycle, If I have two types of boundary data? Or do I need to
choose between (i) non-tidal forcing in my prescribed OBCS and (ii) tidal
forcing in OB*_u/vTid*File ?

Thank you for the answer and support.

Regards,
-r/K-

------------

This are my data and data.obcs namelist:


# Continuous equation parameters
 &PARM01
 tRef=15*25.,
 sRef=15*32.,
#viscAhGrid = 4*deltaT* viscAh/L**2
 viscAhGrid = 1.E-5,
 viscAz=1.E-4,
 diffKhT=0.E1,
 diffKhS=0.E1,
 diffKzT=0.E-4,
 diffKzS=0.E-4,
 diffKrBL79surf= 1.E-4,
 diffKrBL79deep= 1.E-4,
 viscC4Leith=1.0,
 viscC4Leithd=1.0,
 viscA4GridMax=0.5,
 useAreaViscLength=.TRUE.,
 gravity=9.81,
 rhoConst=1027.,
 rhoConstFresh=1000.,
 implicitFreeSurface=.TRUE.,
 eosType='MDJWF',
 implicitDiffusion=.TRUE.,
 implicitViscosity=.TRUE.,
 tempAdvScheme    = 33,
 saltAdvScheme    = 33,
 allowFreezing=.FALSE.,
 hFacMin=0.5,
 hFacMindz=5.,
# rigidLid=.TRUE.,
# implicitFreeSurface=.FALSE.,
# useCDscheme=.TRUE.,
 no_slip_sides=.TRUE.,
 no_slip_bottom=.FALSE.,
 bottomDragQuadratic = 2.5E-3,
 globalFiles       = .TRUE.,
 useSingleCpuIO    = .TRUE.,
 staggerTimeStep=.TRUE.,
 convertFW2Salt = -1.,
 useRealFreshWaterFlux = .FALSE.,
 vectorInvariantMomentum = .TRUE.,
 exactConserv=.TRUE.,
 readBinaryPrec=32,
 writeBinaryPrec=32,
 debugLevel=2,
 &

# Elliptic solver parameters
 &PARM02
 cg2dMaxIters=200,
 cg2dTargetResidual=1.E-8,
 &

# Time stepping parameters
 &PARM03
 nIter0            = 518400,
 nTimeSteps        = 518400,
 deltaTmom  = 1.,
 deltaTtracer= 1.,
 deltaTClock = 1.,
 abEps = 0.1,
 pChkptFreq = 518400,
 chkptFreq  = 259200,
 dumpFreq   = 432000,
 monitorFreq= 3600,
# periodicExternalForcing=.TRUE.,
# externForcingPeriod=2592000.,
# externForcingCycle=31104000.,
## Forcing for boundary condition files
# periodicExternalForcing=.TRUE.,
## 1/12th M2 tidal period...
# externForcingPeriod = 3720.,
## one tidal period (12 records need to be in OBEuFile, etc)
# externForcingCycle = 44640.,
 &
# Gridding parameters
 &PARM04
 usingSphericalPolarGrid=.TRUE.,
 delZ= 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9., 10.,
        12., 14., 16.,18., 20.,
 xgOrigin=105.37666667,
 ygOrigin=-6.83,
 delXfile='delX32_Jkt',
 delYfile='delY32_Jkt',
&

# Input datasets
 &PARM05
 bathyFile='bathy_midres32.bin',
 hydrogThetaFile='./T.0008985600.data',
 hydrogSaltFile= './S.0008985600_ed.data',
 uVelInitFile = './U.0008985600.data',
 vVelInitFile  = './V.0008985600.data',
#
 &

----

 &OBCS_PARM01
 OB_Iwest=24*0,117*1,20*0,243*1,96*0,
 OB_Ieast=231*0,269*-1,
 OB_Jnorth=76*0,674*-1,
 OBCS_monitorFreq=86400.,
#
 useOrlanskiNorth=.FALSE.,
 useOrlanskiWest=.FALSE.,
 useOrlanskiEast=.FALSE.,
 useOrlanskiSouth=.FALSE.,
 useOBCSsponge=.FALSE.,
 useOBCSbalance=.TRUE.,
 useOBCSprescribe=.TRUE.,
 useOBCStides = .TRUE.,
 OBCS_tidalPeriod= 44714.16519187, 43200.00128695, 45570.05351172, 43092.,
                   86164.07700507, 92949.63570054, 86637.19977165, 96732.,
                   22357.09055112, 21972.02193004, 22569.04244845,
46459.34692034,
#
 OBNuFile='UVEL_HYCOM.OBCS_N',
 OBNvFile='VVEL_HYCOM.OBCS_N_balance',
 OBNtFile='THETA_HYCOM.OBCS_N',
 OBNsFile='SALT_HYCOM.OBCS_N',
#
 OBSuFile='',
 OBSvFile='',
 OBStFile='',
 OBSsFile='',
#
 OBWuFile='UVEL_HYCOM.OBCS_W_balance',
 OBWvFile='VVEL_HYCOM.OBCS_W',
 OBWtFile='THETA_HYCOM.OBCS_W',
 OBWsFile='SALT_HYCOM.OBCS_W',
#
 OBEuFile='UVEL_HYCOM.OBCS_E_balance',
 OBEvFile='VVEL_HYCOM.OBCS_E',
 OBEtFile='THETA_HYCOM.OBCS_E',
 OBEsFile='SALT_HYCOM.OBCS_E',
#
#
 OBN_vTidamFile='VAMP_TPXO9v5_12Ced01_N.bin',
 OBN_vTidphFile='VPHA_TPXO9v5_12Ced01_N.bin',
 OBN_uTidamFile='UAMP_TPXO9v5_12Ced01_N.bin',
 OBN_uTidphFile='UPHA_TPXO9v5_12Ced01_N.bin',
#
 OBE_uTidamFile='UAMP_TPXO9v5_12Ced01_E.bin',
 OBE_uTidphFile='UPHA_TPXO9v5_12Ced01_E.bin',
 OBE_vTidamFile='VAMP_TPXO9v5_12Ced01_E.bin',
 OBE_vTidphFile='VPHA_TPXO9v5_12Ced01_E.bin',
#
 OBW_uTidamFile='UAMP_TPXO9v5_12Ced01_W.bin',
 OBW_uTidphFile='UPHA_TPXO9v5_12Ced01_W.bin',
 OBW_vTidamFile='VAMP_TPXO9v5_12Ced01_W.bin',
 OBW_vTidphFile='VPHA_TPXO9v5_12Ced01_W.bin',
#
 OBS_uTidamFile='',
 OBS_uTidphFile='',
 OBS_vTidamFile='',
 OBS_vTidphFile='',
 &







On Sat, May 18, 2024 at 10:33 PM Jean-Michel Campin <
***@***.***> wrote:

> Merged #752
> <https://urldefense.com/v3/__https://github.com/MITgcm/MITgcm/pull/752__;!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P0HdaN0S$>
> into master.
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/MITgcm/MITgcm/pull/752*event-12856141300__;Iw!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P4nhTooZ$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AFL23UT6QIMQ6EPFBBKW6QLZC5YCRAVCNFSM6AAAAAA2TCLZQWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJSHA2TMMJUGEZTAMA__;!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P8km3VJO$>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Aditya R. Kartadikaria (RK), Dr. Eng.
Oceanography Research Group,
Faculty of Earth Sciences and Technology,
Bandung Institute of Technology (ITB),

-- 

This message and its contents, including attachments are intended solely 
for the original recipient. If you are not the intended recipient or have 
received this message in error, please notify me immediately and delete 
this message from your computer system. Any unauthorized use or 
distribution is prohibited. Please consider the environment before printing 
this email.

kartadikaria commented:
It turns that I can use deltaT < 1 if i turn off useCAL and useEXF to
FALSE. Moreover I can use useOBCStides =.TRUE. and useOBCSprescribe=.TRUE.
together with deltaT< 1.

Thank you for your time.

Cheers,
-r/K-

On Fri, Mar 13, 2026 at 6:08 AM -r/K- (Aditya R. Kartadikaria) <
***@***.***> wrote:

> Dear All,
>
> Can I set useEXF=.FALSE., but I still use useOBCStides =.TRUE.
> and useOBCSprescribe=.TRUE. in my data.obcs? I need to reduce CFL by
> reducing deltaT < 1. Since data.exf can't use deltaT < 1, then I turn off
> data.exf. However, I still want to read my prescribed OBCS and tides OBCS
> data in data.obcs. Is that possible?
>
> How to set periodicExternalForcing, externForcingPeriod and
> externForcingCycle, If I have two types of boundary data? Or do I need to
> choose between (i) non-tidal forcing in my prescribed OBCS and (ii) tidal
> forcing in OB*_u/vTid*File ?
>
> Thank you for the answer and support.
>
> Regards,
> -r/K-
>
> ------------
>
> This are my data and data.obcs namelist:
>
>
> # Continuous equation parameters
>  &PARM01
>  tRef=15*25.,
>  sRef=15*32.,
> #viscAhGrid = 4*deltaT* viscAh/L**2
>  viscAhGrid = 1.E-5,
>  viscAz=1.E-4,
>  diffKhT=0.E1,
>  diffKhS=0.E1,
>  diffKzT=0.E-4,
>  diffKzS=0.E-4,
>  diffKrBL79surf= 1.E-4,
>  diffKrBL79deep= 1.E-4,
>  viscC4Leith=1.0,
>  viscC4Leithd=1.0,
>  viscA4GridMax=0.5,
>  useAreaViscLength=.TRUE.,
>  gravity=9.81,
>  rhoConst=1027.,
>  rhoConstFresh=1000.,
>  implicitFreeSurface=.TRUE.,
>  eosType='MDJWF',
>  implicitDiffusion=.TRUE.,
>  implicitViscosity=.TRUE.,
>  tempAdvScheme    = 33,
>  saltAdvScheme    = 33,
>  allowFreezing=.FALSE.,
>  hFacMin=0.5,
>  hFacMindz=5.,
> # rigidLid=.TRUE.,
> # implicitFreeSurface=.FALSE.,
> # useCDscheme=.TRUE.,
>  no_slip_sides=.TRUE.,
>  no_slip_bottom=.FALSE.,
>  bottomDragQuadratic = 2.5E-3,
>  globalFiles       = .TRUE.,
>  useSingleCpuIO    = .TRUE.,
>  staggerTimeStep=.TRUE.,
>  convertFW2Salt = -1.,
>  useRealFreshWaterFlux = .FALSE.,
>  vectorInvariantMomentum = .TRUE.,
>  exactConserv=.TRUE.,
>  readBinaryPrec=32,
>  writeBinaryPrec=32,
>  debugLevel=2,
>  &
>
> # Elliptic solver parameters
>  &PARM02
>  cg2dMaxIters=200,
>  cg2dTargetResidual=1.E-8,
>  &
>
> # Time stepping parameters
>  &PARM03
>  nIter0            = 518400,
>  nTimeSteps        = 518400,
>  deltaTmom  = 1.,
>  deltaTtracer= 1.,
>  deltaTClock = 1.,
>  abEps = 0.1,
>  pChkptFreq = 518400,
>  chkptFreq  = 259200,
>  dumpFreq   = 432000,
>  monitorFreq= 3600,
> # periodicExternalForcing=.TRUE.,
> # externForcingPeriod=2592000.,
> # externForcingCycle=31104000.,
> ## Forcing for boundary condition files
> # periodicExternalForcing=.TRUE.,
> ## 1/12th M2 tidal period...
> # externForcingPeriod = 3720.,
> ## one tidal period (12 records need to be in OBEuFile, etc)
> # externForcingCycle = 44640.,
>  &
> # Gridding parameters
>  &PARM04
>  usingSphericalPolarGrid=.TRUE.,
>  delZ= 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9., 10.,
>         12., 14., 16.,18., 20.,
>  xgOrigin=105.37666667,
>  ygOrigin=-6.83,
>  delXfile='delX32_Jkt',
>  delYfile='delY32_Jkt',
> &
>
> # Input datasets
>  &PARM05
>  bathyFile='bathy_midres32.bin',
>  hydrogThetaFile='./T.0008985600.data',
>  hydrogSaltFile= './S.0008985600_ed.data',
>  uVelInitFile = './U.0008985600.data',
>  vVelInitFile  = './V.0008985600.data',
> #
>  &
>
> ----
>
>  &OBCS_PARM01
>  OB_Iwest=24*0,117*1,20*0,243*1,96*0,
>  OB_Ieast=231*0,269*-1,
>  OB_Jnorth=76*0,674*-1,
>  OBCS_monitorFreq=86400.,
> #
>  useOrlanskiNorth=.FALSE.,
>  useOrlanskiWest=.FALSE.,
>  useOrlanskiEast=.FALSE.,
>  useOrlanskiSouth=.FALSE.,
>  useOBCSsponge=.FALSE.,
>  useOBCSbalance=.TRUE.,
>  useOBCSprescribe=.TRUE.,
>  useOBCStides = .TRUE.,
>  OBCS_tidalPeriod= 44714.16519187, 43200.00128695, 45570.05351172, 43092.,
>                    86164.07700507, 92949.63570054, 86637.19977165, 96732.,
>                    22357.09055112, 21972.02193004, 22569.04244845,
> 46459.34692034,
> #
>  OBNuFile='UVEL_HYCOM.OBCS_N',
>  OBNvFile='VVEL_HYCOM.OBCS_N_balance',
>  OBNtFile='THETA_HYCOM.OBCS_N',
>  OBNsFile='SALT_HYCOM.OBCS_N',
> #
>  OBSuFile='',
>  OBSvFile='',
>  OBStFile='',
>  OBSsFile='',
> #
>  OBWuFile='UVEL_HYCOM.OBCS_W_balance',
>  OBWvFile='VVEL_HYCOM.OBCS_W',
>  OBWtFile='THETA_HYCOM.OBCS_W',
>  OBWsFile='SALT_HYCOM.OBCS_W',
> #
>  OBEuFile='UVEL_HYCOM.OBCS_E_balance',
>  OBEvFile='VVEL_HYCOM.OBCS_E',
>  OBEtFile='THETA_HYCOM.OBCS_E',
>  OBEsFile='SALT_HYCOM.OBCS_E',
> #
> #
>  OBN_vTidamFile='VAMP_TPXO9v5_12Ced01_N.bin',
>  OBN_vTidphFile='VPHA_TPXO9v5_12Ced01_N.bin',
>  OBN_uTidamFile='UAMP_TPXO9v5_12Ced01_N.bin',
>  OBN_uTidphFile='UPHA_TPXO9v5_12Ced01_N.bin',
> #
>  OBE_uTidamFile='UAMP_TPXO9v5_12Ced01_E.bin',
>  OBE_uTidphFile='UPHA_TPXO9v5_12Ced01_E.bin',
>  OBE_vTidamFile='VAMP_TPXO9v5_12Ced01_E.bin',
>  OBE_vTidphFile='VPHA_TPXO9v5_12Ced01_E.bin',
> #
>  OBW_uTidamFile='UAMP_TPXO9v5_12Ced01_W.bin',
>  OBW_uTidphFile='UPHA_TPXO9v5_12Ced01_W.bin',
>  OBW_vTidamFile='VAMP_TPXO9v5_12Ced01_W.bin',
>  OBW_vTidphFile='VPHA_TPXO9v5_12Ced01_W.bin',
> #
>  OBS_uTidamFile='',
>  OBS_uTidphFile='',
>  OBS_vTidamFile='',
>  OBS_vTidphFile='',
>  &
>
>
>
>
>
>
>
> On Sat, May 18, 2024 at 10:33 PM Jean-Michel Campin <
> ***@***.***> wrote:
>
>> Merged #752
>> <https://urldefense.com/v3/__https://github.com/MITgcm/MITgcm/pull/752__;!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P0HdaN0S$>
>> into master.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://urldefense.com/v3/__https://github.com/MITgcm/MITgcm/pull/752*event-12856141300__;Iw!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P4nhTooZ$>,
>> or unsubscribe
>> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AFL23UT6QIMQ6EPFBBKW6QLZC5YCRAVCNFSM6AAAAAA2TCLZQWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJSHA2TMMJUGEZTAMA__;!!Nmw4Hv0!xH1_wgq3gfNyukt061_i_W8Qw0maNcDw_58ToMNjyhzStfY89uRW03Ab-g-G14BoQ1gBH3lubu-VuqOR_I3WTpEcSHc_P8km3VJO$>
>> .
>> You are receiving this because you are subscribed to this thread.Message
>> ID: ***@***.***>
>>
>
>
> --
> Aditya R. Kartadikaria (RK), Dr. Eng.
> Oceanography Research Group,
> Faculty of Earth Sciences and Technology,
> Bandung Institute of Technology (ITB),
>
>
>

-- 

This message and its contents, including attachments are intended solely 
for the original recipient. If you are not the intended recipient or have 
received this message in error, please notify me immediately and delete 
this message from your computer system. Any unauthorized use or 
distribution is prohibited. Please consider the environment before printing 
this email.
